File: Reverse_Prefix_of_Word.go
package main

import (
	"fmt"
)

func main() {
	word := "abcdefd"
	ch := "d"
	fmt.Println(reversePrefix(word, ch))
}

func reversePrefix(word string, ch string) string {
	for i, s := range word {
		if string(s) == string(ch) {
			return reverse(word[:i+1]) + word[i+1:]
		}
	}

	return word
}

func reverse(w1 string) (result string) {
	for _, s1 := range w1 {
		result = string(s1) + result
	}
	return
}

File: Reverse_Prefix_of_Word/Reverse_Prefix_of_Word.py
word = "abcdefd"
ch = "d"
# word = "abcd"
# ch = "z"
def reversePrefix(word,ch):
    w1 = list(word)
   
    for i in w1:
        if i == ch:
            idx = w1.index(ch)+1
            return "".join(w1[:idx][::-1]+w1[idx:])
        
    return "".join(w1)


print(reversePrefix(word,ch))